docs/start-notebook-and-manage-dependencies.md

```diff
@@ -0,0 +1,228 @@
+# Reproducibility of Jupyter Notebooks
+
+Reproducibility and shareability of notebooks is very important if you want to allow others to repeat your experiments and avoid issues due to dependencies management.
+When using `pip install <package_name>` is not possible to verify which software stack was used to run the notebook and therefore another user cannot repeat the same experiment.
+Dependency management is one of the most important requirements for reproducibility. Having dependencies clearly stated allows portability of notebooks, so they can be shared safely with others, reused in other projects or simply reproduced. If you want to know more about this issue in the data science domain, have a look at this [article](https://developers.redhat.com/blog/2021/03/19/managing-python-dependencies-with-the-thoth-jupyterlab-extension/) or this [video](https://www.youtube.com/watch?v=ifyQ2oSxjnU).
+
+In order to help developers (including data scientists), dependencies for Jupyter notebooks in this tutorial are managed using the JupyterLab extension [jupyterlab-requirements][1].
+
+You can use this extension for each of your notebook to guarantee they have the correct dependencies. This extension is able to add/remove dependencies, lock them and store them in the notebook metadata. In this way all the dependencies information required to repeat the environment are shipped with the notebook.
+
+In particular, in the notebook metadata you will find:
+
+- `requirements (Pipfile)`;
+
+- `requirements lock with all versions and hashes (Pipfile.lock)`;
+
+- `dependency resolution engine` used (Thoth or Pipenv);
+
+- `.thoth.yaml configuration file` (only for Thoth resolution engine).
+
+All this information can allow reproducibility and shareability of the notebook.
+
+
+## Manage dependencies with the jupyterlab-requirements extension
+
+There are 3 ways to interact with [jupyterlab-requirements][1] JupyterLab extension:
+
+- using `%horus` magic commands directly in your notebook's cells (preferred approach). To learn more about how to use the `%horus` magic commands check out the guide [here](https://github.com/thoth-station/jupyterlab-requirements#horus-magic-command) or the video [here](https://www.youtube.com/watch?v=FjVxNTXO70I)
+
+<div style="text-align:center">
+<img alt="JupyterLab Requirements Horus magic commands" src="https://raw.githubusercontent.com/AICoE/manage-dependencies-tutorial/master/docs/images/JupyterLabRequirementsExtensionMC.png">
+</div>
+
+- using the `horus` CLI directly from terminal or integrated in pipelines ([check video](https://www.youtube.com/watch?v=fW0YKugL26g&t)).
```

Comment:
is there a documentation on this that we can link?

Reply:
https://github.com/thoth-station/jupyterlab-requirements#horus-jupyterlab-requirements-cli